docs/src/main/asciidoc/keycloak-authorization-guide.adoc

```diff
@@ -0,0 +1,269 @@
+
+[source, subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=using-keycloak-authorization \
+    -Dextensions="oidc, resteasy-jsonb"
+----
+
+This command generates a Maven project, importing the `keycloak` extension
+which is an implementation of a Keycloak Adapter for Quarkus applications and provides all the necessary capabilities to integrate with a Keycloak Server and perform bearer token authorization.
+
+== Writing the application
+
+Let's start by implementing the `/api/users/me` endpoint. As you can see from the source code below it is just a regular JAX-RS resource:
+
+[source,java]
+----
+package org.acme.keycloak;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.jboss.resteasy.annotations.cache.NoCache;
+
+import io.quarkus.security.identity.SecurityIdentity;
+
+/**
+ * @author <a href="mailto:psilva@redhat.com">Pedro Igor</a>
+ */
+@Path("/api/users")
+public class UsersResource {
+
+    @Inject
+    SecurityIdentity identity;
+
+    @GET
+    @Path("/me")
+    @Produces(MediaType.APPLICATION_JSON)
+    @NoCache
+    public User me() {
+        return new User(identity);
+    }
+
+    public class User {
+
+        private final String userName;
+
+        User(SecurityIdentity identity) {
+            this.userName = identity.getPrincipal().getName();
+        }
+
+        public String getUserName() {
+            return userName;
+        }
+    }
+}
+----
+
+The source code for the `/api/admin` endpoint is also very simple:
+
+[source,java]
+----
+package org.acme.keycloak;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import io.quarkus.security.Authenticated;
+
+/**
+ * @author <a href="mailto:psilva@redhat.com">Pedro Igor</a>
+ */
+@Path("/api/admin")
+@Authenticated
+public class AdminResource {
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String admin() {
+        return "granted";
+    }
+}
+----
+
+Note that we did not define any annotation such as `RoleAllowed` to explicitly enforce access to a resource. The extension will
+be responsible to map the URIs of the protected resources you have in Keycloak and evaluate the permissions accordingly, granting or denying access depending on the permissions that will be granted by Keycloak.
+
+== Configuring the application
+
+The OpenID Connect extension allows you to define the adapter configuration using the `application.properties` file which should be located at the `src/main/resources` directory.
+
+=== Configuring using the application.properties file
+
+[source,properties]
+----
+# OIDC Configuration
+quarkus.oidc.auth-server-url=http://localhost:8180/auth/realms/quarkus
+quarkus.oidc.client-id=backend-service
+quarkus.oidc.credentials.secret=secret
```

Comment:
Is this `secret` actually used by the enforcer ? Just checking, it was not used in the `integration-tests/oidc`, but may be the enforcer does need it